custom_components/ha-google-home/api.py

```diff
@@ -86,28 +86,42 @@ def get_alarms_and_timers_from(self, device, endpoint):
     def get_google_devices_information(self):
         """Retrieves devices from glocaltokens"""
         _LOGGER.debug("Fetching data...")
+        offline_devices = []
         devices = self._client.get_google_devices()
 
         for device in devices:
             # To avoid key error's
             timers = []
             alarms = []
-
-            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
-            if result:
-                timers = result.get(TIMERS)
-                alarms = result.get(ALARMS)
-
-                if not timers and not alarms:
-                    _LOGGER.error(
-                        "For device {device} - {error}".format(
-                            device=device.device_name, error=API_RETURNED_UNKNOWN
+            if device.ip:
+                result_json = self.get_alarms_and_timers_from(
+                    device, API_ENDPOINT_ALARMS
+                )
+                if result_json:
+                    timers = result_json.get(TIMERS)
+                    alarms = result_json.get(ALARMS)
+
+                    if not timers and not alarms:
+                        _LOGGER.error(
+                            "For device {device} - {error}".format(
+                                device=device.device_name, error=API_RETURNED_UNKNOWN
+                            )
                         )
-                    )
+            else:
+                offline_devices.append(device.device_name)
+                device.timers = timers
+                device.alarms = alarms
 
-            device.timers = timers
-            device.alarms = alarms
             _LOGGER.debug(device)
 
+        # Gives the user a warning if the device is offline, but will not remove entities or device from
+        # HA device registry
+        if offline_devices:
+            for device in offline_devices:
+                _LOGGER.warning(
+                    "Failed to fetch timers/alarms information from device {device}. Will try again later.".format(
+                        device=device
+                    )
+                )
```

Comment:
Since you have `device` objects inside your list, just print the name here, if you wish
```
LOGGER.warning(
"Failed to fetch timers/alarms information from device {device_name}. Will try again later.".format(device_name=device.device_name)
)
```